server/src/test/java/org/opensearch/indices/replication/SegmentReplicationTargetServiceTests.java

```diff
@@ -242,7 +244,46 @@ public void testAlreadyOnNewCheckpoint() {
     }
 
     public void testShardAlreadyReplicating() {
-        sut.startReplication(replicaShard, mock(SegmentReplicationTargetService.SegmentReplicationListener.class));
+        CountDownLatch blockGetCheckpointMetadata = new CountDownLatch(1);
+        SegmentReplicationSource source = new TestReplicationSource() {
+            @Override
+            public void getCheckpointMetadata(
+                long replicationId,
+                ReplicationCheckpoint checkpoint,
+                ActionListener<CheckpointInfoResponse> listener
+            ) {
+                try {
+                    blockGetCheckpointMetadata.await();
+                    final CopyState copyState = new CopyState(
+                        ReplicationCheckpoint.empty(primaryShard.shardId(), primaryShard.getLatestReplicationCheckpoint().getCodec()),
+                        primaryShard
+                    );
+                    listener.onResponse(
+                        new CheckpointInfoResponse(copyState.getCheckpoint(), copyState.getMetadataMap(), copyState.getInfosBytes())
+                    );
+                } catch (InterruptedException | IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+
+            @Override
+            public void getSegmentFiles(
+                long replicationId,
+                ReplicationCheckpoint checkpoint,
+                List<StoreFileMetadata> filesToFetch,
+                IndexShard indexShard,
+                ActionListener<GetSegmentFilesResponse> listener
+            ) {
+                listener.onResponse(new GetSegmentFilesResponse(Collections.emptyList()));
+            }
+        };
+        final SegmentReplicationTarget target = spy(
+            new SegmentReplicationTarget(replicaShard, source, mock(SegmentReplicationTargetService.SegmentReplicationListener.class))
+        );
+        // Start first round of segment replication.
+        sut.startReplication(target);
```

Comment:
The source before this is already a mock, 
```
        replicationSource = mock(SegmentReplicationSource.class);
        when(replicationSourceFactory.get(replicaShard)).thenReturn(replicationSource);
```
curious how is it completing a round here with an empty mock? The listeners shouldn't get resolved.

Reply:
ReplicationSource is driven from `SegmentReplicationSourceFactory` which is defined in IndexShardTestCase [prepareForReplication](https://github.com/opensearch-project/OpenSearch/blob/3c973bafe709565380eb4224242e860499fb4acc/test/framework/src/main/java/org/opensearch/index/shard/IndexShardTestCase.java#L1350). If we want to add mock `ReplicationSource` we can define a new method. But, I don't see a strong need as we are already handing it inside IndexShardTestCase [here](https://github.com/opensearch-project/OpenSearch/blob/3c973bafe709565380eb4224242e860499fb4acc/test/framework/src/main/java/org/opensearch/index/shard/IndexShardTestCase.java#L1454)